src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js

```diff
@@ -643,7 +648,7 @@
         });
 
         // update navigation menu
-        if (cfg.navigate_menu) {
+        if (cfg.navigate_menu && liveNotebook) {
```

Comment:
I agree with that. That is the reason of the issue you encountered: if the navigate menu is enabled in the live notebook, then this config is stored in the notebook metadata and the js will try to create this menu even in the non live situation. 

Alternatively, we could also fix in hard the value `navigate_menu=false` in the toc2 template `toc2.tpl` which is used by nbconvert.